app/src/main/java/com/github/sdp/mediato/CreateProfileFragment.java

```diff
@@ -0,0 +1,136 @@
+
+        //Set an error if the password is less than 8 characters.
+        createButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                int error = 0;
+                String errorMsg;
+                if (0 != (error = isPasswordValid(usernameEditText.getText()))) {
+                    errorMsg = (error == 1) ? getString(R.string.mt_username_error_1)
+                            : (error == 2) ? getString(R.string.mt_username_error_2)
+                            : getString(R.string.mt_username_error_3);
+
+                    usernameTextInput.setError(errorMsg);
+                } else {
+                    usernameTextInput.setError(null);
+                    //((NavigationHost) getActivity()).navigateTo(new ProductGridFragment(), false);
+                }
+            }
+        });
+
+        // Inflate the layout for this fragment
+        return view;
+    }
+
+    @NonNull
+    private View.OnKeyListener removeErrorIfUsernameValid(TextInputLayout usernameTextInput, TextInputEditText usernameEditText) {
+        return new View.OnKeyListener() {
+            @Override
+            public boolean onKey(View view, int i, KeyEvent keyEvent) {
+                if (0 == isPasswordValid(usernameEditText.getText())) {
+                    usernameTextInput.setError(null);
+                }
+                return false;
+            }
+        };
+    }
+
+    @NonNull
+    private View.OnClickListener generateUsername(TextInputLayout usernameTextInput, TextInputEditText usernameEditText) {
+        return new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                usernameTextInput.setError(null);
+                Faker faker = new Faker();
+                String username = faker.animal().name();
+                String number = faker.number().digits(5);
+                usernameEditText.setText(username + number);
+            }
+        };
+    }
+
+    @NonNull
+    private View.OnClickListener photoPicker() {
+        return new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ImagePicker.Companion.with(CreateProfileFragment.this)
+                        .crop()
+                        .cropSquare()
+                        .compress(1024)
+                        .maxResultSize(620, 620)
+                        .start();
+            }
+        };
+    }
+
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        assert data != null;
+        Uri uri = data.getData();
+        profileImage.setImageURI(uri);
+    }
+
+    // "isPasswordValid" from "Navigate to the next Fragment" section method goes here
+    private int isPasswordValid(@Nullable Editable text) {
```

Comment:
Can you explain how this method works ? I think we can find a cleaner way to make the return values more understandable (maybe an enum for the error type)

Reply:
I have change that